src/main/kotlin/com/team4099/robot2020/subsystems/Intake.kt

```diff
@@ -0,0 +1,93 @@
+package com.team4099.robot2020.subsystems
+
+import com.revrobotics.CANSparkMax
+import com.revrobotics.CANSparkMaxLowLevel.MotorType
+import com.team4099.lib.logging.HelixLogger
+import com.team4099.lib.motorcontroller.SparkMaxControllerFactory
+import com.team4099.lib.subsystem.Subsystem
+import com.team4099.robot2020.config.Constants
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard
+
+object Intake : Subsystem {
+
+    private val sparkMax = CANSparkMax(Constants.Intake.INTAKE_SPARK_MAX_ID, MotorType.kBrushless)
+
+    private var intakePower = 0.0
+    var intakeState = IntakeState.IDLE
+
+    enum class IntakeState {
+        IN, IDLE, OUT
+    }
+
+    fun outputToSmartDashboard() {
+        SmartDashboard.putString("intake/intakeState", intakeState.toString())
+    }
+
+    init {
+        sparkMax.inverted = false
+    }
+
+    @Synchronized
+
+    private fun setIntakePower(power: Double) {
+        sparkMax.set(power)
+    }
+
+
+    override fun outputTelemetry() {
+        SmartDashboard.putString("intake/intakeState", intakeState.toString())
+        SmartDashboard.putNumber("intake/intakePower", intakePower)
+    }
+
+    override fun checkSystem() {}
+
+    override fun registerLogging() {
+        HelixLogger.addSource("Intake motor power") { sparkMax.outputCurrent}
+    }
```

Comment:
name doesn't describe what's actually happening here, should be intake output current. also add state and actual power here

Reply:
we didnt merge that hotfix into this branch

Reply:
which hotfix 🤔